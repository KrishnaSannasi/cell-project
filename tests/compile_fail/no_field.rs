use cell_project::cell_project as project;
use std::cell::Cell;

pub struct Foo {
    bar: i32,
    quax: i32,
}

fn project(cell: &Cell<Foo>) -> &Cell<i32> {
    project!(Foo, cell.foo)
}

fn main() {}
