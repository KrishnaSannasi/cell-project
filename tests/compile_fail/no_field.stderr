error[E0026]: struct `Foo` does not have a field named `foo`
  --> $DIR/no_field.rs:10:24
   |
10 |     project!(Foo, cell.foo)
   |                        ^^^ struct `Foo` does not have this field
